### PR TITLE
30 add budgets to invoices UI

### LINKED ISSUE #30: Add budgets to invoices UI
Currently, users are able to approve invoices in the UI, but don't have contextual reference to the budget of the relevant department or project. 

We need to improve the invoice approvals UI so that relevant budget data is pulled from Odoo, and displayed to the user so they can understand the exact impact of their approval. 

Budgets in Odoo are not updated right away, so we will need to add a logic that enables the user to see a running total for their session.

### PR DESCRIPTION
**Pull request by lewisdonovan**
# Add Budget Impact Analysis

## Overview
This commit implements a comprehensive budget impact analysis system for invoice approval workflows, enabling users to see real-time budget implications before approving invoices.

### 🎯 Budget Impact Analysis
- **Real-time Budget Calculations**: Shows remaining budget and impact for both construction and department invoices
- **Construction Invoice Support**: Analyzes project-specific budgets with spend category breakdowns
- **Department Invoice Support**: Calculates monthly department budget utilization
- **Visual Indicators**: Clear UI showing budget status with color-coded warnings for over-budget scenarios

### 📊 Budget Context System
- **BudgetContext**: Centralized state management for budget calculations across components
- **Session Tracking**: Tracks approved invoices in current session to show cumulative impact
- **Real-time Updates**: Budget impact updates automatically when selections change

### 🔧 Odoo Integration
- **Budget Data Fetching**: Integrates with Odoo's `account.report.budget` and `account.report.budget.item` models
- **Invoice Line Analysis**: Analyzes `account.move.line` records for accurate budget tracking
- **Analytic Distribution**: Properly handles Odoo's analytic accounting for project/department allocation

### 🎨 User Interface
- **BudgetImpactCard Component**: Dedicated component showing budget analysis with clear visual indicators
- **Internationalization**: Full i18n support for English and Spanish
- **Responsive Design**: Works across all screen sizes with proper dark mode support

### New API Routes
- `/api/budgets/[analyticAccountId]` - Fetch budget data for specific analytic account
- `/api/budgets/impact/construction` - Calculate construction budget impact
- `/api/budgets/impact/department` - Calculate department budget impact
- `/api/budgets/impact` - General budget impact calculation

### Enhanced Services
- **Budget Calculation Functions**: `calculateConstructionBudgetImpact`, `calculateDepartmentBudgetImpact`
- **Odoo Data Fetching**: `getBudgetForProject`, `getBudgetForDepartment`, `getBudgetLineItem`
- **Invoice Analysis**: `getApprovedInvoicesForProjectAndCategory`, `getApprovedInvoicesForDepartmentInMonth`

### Type System Updates
- **New Types**: `OdooBudget`, `OdooBudgetLineItem`, `BudgetImpact`, `OdooInvoiceLineItem`
- **Enhanced Interfaces**: Updated `OdooInvoice` with additional fields for budget analysis

## Files Added/Modified
- **New Components**: `BudgetImpactCard.tsx`, `BudgetContext.tsx`
- **New Services**: `budgetSessionService.ts`
- **API Routes**: 4 new budget-related endpoints
- **Enhanced Pages**: Invoice detail page with budget analysis integration
- **Internationalization**: Added budget-related translation keys

## User Experience
1. User selects department/project on invoice detail page
2. Budget impact analysis automatically calculates and displays
3. Clear visual indicators show remaining budget and potential overages
4. Session tracking shows cumulative impact of multiple approvals
5. Real-time updates as user changes selections

## Testing
- Added comprehensive test documentation in `budget-integration.test.md`
- Covers both construction and department invoice scenarios
- Includes edge cases and error handling validation

## Impact
- **20 files changed**: 2,117 additions, 54 deletions
- **Enhanced Invoice Workflow**: Users can now make informed decisions about budget impact
- **Improved Accuracy**: Real-time budget calculations prevent over-budget approvals
- **Better UX**: Clear visual feedback and intuitive budget analysis interface

Closes #30

### CODE REVIEW

File: apps/kpis/src/lib/odoo/services.ts [hunk 1]
```diff
@@ -597,4 +703,588 @@ export async function approveInvoice(invoiceId: number, departmentId?: number, p
...
+        
+        // Get budget for project
+        const budget = await getBudgetForProject(projectId);
+        if (!budget) {
+            console.warn(`No budget found for project ${projectId}`);
+            return null;
+        }
+
+        // Get budget line item for spend category
+        const lineItem = await getBudgetLineItem(budget.id, spendCategoryId.toString());
```
[the remark this answers is not in the document]
Comment (lewisdonovan) [reply apps/kpis/src/lib/odoo/services.ts]: Resolved in 2e13d439c02c81d77adfd112b5bc1a3b1e9a5b58
